demos/ping_pong.py

```diff
@@ -0,0 +1,233 @@
+# :math:`20\times25` with diagonals from top left to bottom right and mesh B
+# :math:`20\times20` with diagonals from bottom left to top right. ::
+
+mesh_A = UnitSquareMesh(20, 25, diagonal="left")
+mesh_B = UnitSquareMesh(20, 20, diagonal="right")
+
+fig, axes = plt.subplots(ncols=2, figsize=(10, 5))
+for i, mesh in enumerate((mesh_A, mesh_B)):
+    triplot(mesh, axes=axes[i])
+    axes[i].set_title(f"Mesh {'B' if i else 'A'}")
+    axes[i].axis(False)
+    axes[i].set_aspect(1)
+plt.savefig("ping_pong-meshes.jpg")
+
+# Define the sensor function
+#
+# ..math::
+#   f(x,y) = \sin(\pi x) \sin(\pi y).
+#
+# Let's plot the sensor function as represented in :math:`\mathbb{P}1` spaces on mesh A.
+# ::
+
+V_A = FunctionSpace(mesh_A, "CG", 1)
+V_B = FunctionSpace(mesh_B, "CG", 1)
+x, y = SpatialCoordinate(mesh_A)
+sensor = Function(V_A).interpolate(sin(pi * x) * sin(pi * y))
+
+fig, axes = plt.subplots()
+tricontourf(sensor, axes=axes)
+axes.set_title("Source function")
+axes.axis(False)
+axes.set_aspect(1)
+plt.savefig("ping_pong-source_function.jpg")
+
+# To start with, let's consider the `interpolate` approach, which point evaluates the
+# input field at the locations where degrees of freedom of the target function space.
+# We run the experiment for 50 iterations and track three quantities as the iterations
+# progress: the integral of the field, its global minimum, and its global maximum. ::
+
+niter = 1 if os.environ.get("ANIMATE_REGRESSION_TEST") else 50
+initial_integral = assemble(sensor * dx)
+initial_min = sensor.vector().gather().min()
+initial_max = sensor.vector().gather().max()
+quantities = {
+    "integral": {"interpolate": [initial_integral]},
+    "minimum": {"interpolate": [initial_min]},
+    "maximum": {"interpolate": [initial_max]},
+}
+f_interp = Function(V_A).assign(sensor)
+for _ in range(niter):
+    f_interp = interpolate(interpolate(f_interp, V_B), V_A)
+    quantities["integral"]["interpolate"].append(assemble(f_interp * dx))
+    quantities["minimum"]["interpolate"].append(f_interp.vector().gather().min())
+    quantities["maximum"]["interpolate"].append(f_interp.vector().gather().max())
+
+fig, axes = plt.subplots(ncols=3, figsize=(15, 5))
+for i, (key, subdict) in enumerate(quantities.items()):
+    axes[i].plot(subdict["interpolate"])
+    axes[i].set_xlabel("Iteration")
+    axes[i].set_ylabel(key.capitalize())
+    axes[i].grid(True)
+plt.savefig("ping_pong-quantities_interpolate.jpg")
+
+# .. figure:: ping_pong-quantities_interpolate.jpg
+#    :figwidth: 90%
+#    :align: center
+#
+# The first plot shows the integral of the field (the 'mass') as the interpolation
+# iterations progress. Note that the mass drops steeply, before leveling out after
+# around 50 iterations. Minimum values are perfectly maintained for this example,
+# staying at zero for all iterations. Maximum values, however, decrease for several
+# iterations before leveling out after around 30 iterations.
```

Comment:
I think it's useful to say here what you said on the meeting too: that this is due to the minimum values being achieved at the boundary, while the maximum is in the middle (and then 20x20 <-> 20x25 resolution shifting that)